source/common/upstream/load_balancer_impl.h

```diff
@@ -74,13 +76,33 @@ class LoadBalancerBase {
   /**
    * Regenerate locality aware routing structures for fast decisions on upstream locality selection.
    */
-  void regenerateLocalityRoutingStructures();
-
-  const HostSet* local_host_set_;
-  uint64_t local_percent_to_route_{};
-  LocalityRoutingState locality_routing_state_{LocalityRoutingState::NoLocalityRouting};
-  std::vector<uint64_t> residual_capacity_;
-  Common::CallbackHandle* local_host_set_member_update_cb_handle_{};
+  void regenerateLocalityRoutingStructures(uint32_t priority);
+
+  uint32_t best_available_priority() { return best_available_host_set_->priority(); }
```

Comment:
Nit, this and other methods might be `cont` candidates (for their side effects, not return).